File: Java/Digit-Counts.java
class Solution {
    /*
     * param k : As description.
     * param n : As description.
     * return: An integer denote the count of digit k in 1..n
     */
    public int digitCounts(int k, int n) {
        // write your code here
        int res = 0;
        for (int i = 0; i <= n; i ++) {
            int cur = i;
            do {
                if (cur % 10 == k) res ++;
                cur = cur / 10;
            } while (cur > 0);
        }
        return res;
    }
};

File: Python/Digit-Counts.py
class Solution:
    # @param k & n  two integer
    # @return ans a integer
    def digitCounts(self, k, n):
        res = 0
        if k == 0 and n >= 0:
            res += 1
    	for i in range(1, n + 1):
            cur = i
            while cur != 0:
                if cur % 10 == k:
                    res += 1
                cur = cur / 10
        return res